spring-integration-xmpp/src/main/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpoint.java

```diff
@@ -64,11 +70,37 @@ public void setHeaderMapper(XmppHeaderMapper headerMapper) {
 	 * Specify whether the text message body should be extracted when mapping to a
 	 * Spring Integration Message payload. Otherwise, the full XMPP Message will be
 	 * passed within the payload. This value is <em>true</em> by default.
-	 *
 	 * @param extractPayload true if the payload should be extracted.
+	 * @deprecated since version 4.3 in favor of {@link #setPayloadExpression(Expression)}
 	 */
+	@Deprecated
 	public void setExtractPayload(boolean extractPayload) {
-		this.extractPayload = extractPayload;
+		if (this.payloadExpression == null) {
+			setPayloadExpression(extractPayload ?
+					EXPRESSION_PARSER.parseExpression("payload")
```

Comment:
The `XmppMessage` doesn't have `payload` - shouldn't this be `body` ?

There is no test coverage for this case.